apps/web/src/pages/auth/LoginPage.tsx

```diff
@@ -27,19 +35,45 @@ export default function LoginPage() {
       if (!user.organizationId || !user.environmentId) {
         navigate('/auth/application');
       } else {
+        if (code && next) {
+          startVercelSetup();
+
+          return;
+        }
+
         navigate(source === 'cli' ? '/quickstart' : '/');
       }
     }
   }, [token]);
 
   return (
     <AuthLayout>
-      <AuthContainer
-        title="Sign In"
-        description=" Welcome back! Sign in with the data you entered in your registration"
-      >
-        <LoginForm />
-      </AuthContainer>
+      {isLoading ? (
+        <Paper
+          sx={{
+            display: 'flex',
+            justifyContent: 'center',
+            alignItems: 'center',
+            flexDirection: 'column',
+            minHeight: '100vh',
+            position: 'fixed',
+            top: 0,
+            left: 0,
+            right: 0,
+            bottom: 0,
+          }}
+        >
+          <Loader color={colors.error} size={32} />
+          <Text>Setting up Vercel Integration...</Text>
+        </Paper>
```

Comment:
I would use the `LoadingOverlay` here instead. Keeps it simple

Reply:
I wanted to have the `setting up vercel integration` text as well